.github/workflows/python-setup.yml

```diff
@@ -2,25 +2,32 @@
 
 name: Python Setup
 
-on:
-  push:
-    branches: [ main ]
-  pull_request:
-    branches: [ main ]
+on: [push, pull_request]
 
 env:
-    BLACK_VERSION: 23.12.1
-    PYTHON_VERSION: 3.11
+    PYTHON_VERSION: 3.11.6
+
+permissions:
+  contents: read
 
 jobs:
-  black-format:
+  build:
 
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v3
-    - uses: psf/black@stable
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
       with:
-         options: "--check --verbose"
-         src: "./src"
-         version: ${{env.BLACK_VERSION}}         
+        python-version: ${{env.PYTHON_VERSION}}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        python -m pip install pre-commit jupyter
+
+    - name: Install dependencies
```

Comment:
I suggest renaming this to something like 'Install pre-commit hooks' to differentiate it from the previous step when viewing workflow runs.